### ISSUE TITLE
Xcode 7.3 beta 2 & 3 issue

### ISSUE THREAD

**Issue by rusdariusalex** [xmartlabs/Eureka#212]
I get this error when i try to use Eureka in Xcode 7.3 beta 2 (swift 2.2) 

"initializer requirement 'init' can only be satisfied by a `required` initializer in the definition of non-final class 'Section'"

There are also some deprecations

**Comment by hariseldon78** [comment]
did you find a way to compile? the error does not say the file and line...

**Comment by rusdariusalex** [comment]
couldn't compile. i'm using Xcode 7.2 until it will be fixed

**Comment by mtnbarreto** [comment]
I will look into it shortly. Use Xcode 7.2 for now.

**Comment by csotiriou** [comment]
Not sure if it helps ( I didn't have the time to look into it myself ), but looking at the Xcode 7.3 beta 2 release notes, Apple states there are changes made to the initialisers. Could this affect Eureka?

> Inside a class, a designated initializer that is either failable (init?()) or throwing (init() throws) is allowed to exit before initializing all stored properties and calling super.init(). This behavior is supported, making designated initializers more consistent with convenience initializers. Convenience initializers can also fail before performing a self.init() delegation. (18216578)

**Comment by TeunVR** [comment]
Most of the fixes are relatively easy, solving the deprecated -- and ++ operators.
However, the tricky compiler error remains:
<unknown>:0: error: initializer requirement 'init' can only be satisfied by a `required` initializer in the definition of non-final class 'Section'

I think it is caused by the public init() requirement from the RangeReplaceableCollectionType protocol that is added in the extension Section : RangeReplaceableCollectionType.
It looks like this bug: http://www.openradar.me/23358609

If i make the Section class final for testing (and remove SelectableSection), the error goes away but then the compiler segfaults with a stackoverflow :(

**Comment by mtnbarreto** [comment]
I just reported the issue: https://bugs.swift.org/browse/SR-661

**Comment by csotiriou** [comment]
To me, seems like an issue on Apple's side.

Does it happen with beta 3?

**Comment by ffittschen** [comment]
Just tried it with Xcode 7 beta 3, still crashing...:
```
*** Building scheme "Eureka" in Eureka.xcworkspace
2016-02-09 12:12:09.245 xcodebuild[24873:8586812]  DVTAssertions: Warning in /Library/Caches/com.apple.xbs/Sources/IDEFrameworks/IDEFrameworks-10160.3/IDEFoundation/Playgrounds/IDEPlaygroundAuxiliarySourceCompilerOperation.m:312
Details:  Unable to read diagnostics from file "/Users/ffittschen/Library/Developer/Xcode/DerivedData/Eureka-cxulzhapymzunnejqpbsnszfxwdk/Build/Intermediates/Eureka.build/Release-iphoneos/Eureka.build/Objects-normal/arm64/Rows.dia" (Invalid File): Invalid diagnostics signature
Function: void XCGenerateDiagnosticsFromFile(NSString *__strong, NSString *__strong, NSDictionary *__strong, NSDictionary *__strong, IDEActivityLogSectionRecorder *__strong, BOOL (^__strong)(IDEActivityLogMessage *__strong))
Thread:   <NSThread: 0x7fae66d222b0>{number = 6, name = (null)}
Please file a bug at http://bugreport.apple.com with this warning message and any useful information you can provide.
2016-02-09 12:12:09.846 xcodebuild[24873:8586812]  DVTAssertions: Warning in /Library/Caches/com.apple.xbs/Sources/IDEFrameworks/IDEFrameworks-10160.3/IDEFoundation/Playgrounds/IDEPlaygroundAuxiliarySourceCompilerOperation.m:312
Details:  Unable to read diagnostics from file "/Users/ffittschen/Library/Developer/Xcode/DerivedData/Eureka-cxulzhapymzunnejqpbsnszfxwdk/Build/Intermediates/Eureka.build/Release-iphoneos/Eureka.build/Objects-normal/arm64/Cells.dia" (Invalid File): Invalid diagnostics signature
Function: void XCGenerateDiagnosticsFromFile(NSString *__strong, NSString *__strong, NSDictionary *__strong, NSDictionary *__strong, IDEActivityLogSectionRecorder *__strong, BOOL (^__strong)(IDEActivityLogMessage *__strong))
Thread:   <NSThread: 0x7fae66d222b0>{number = 6, name = (null)}
Please file a bug at http://bugreport.apple.com with this warning message and any useful information you can provide.
** BUILD FAILED **


The following build commands failed:
    CompileSwift normal arm64 /Users/ffittschen/Dev/TemplateApp/Carthage/Checkouts/Eureka/Source/Core.swift
    CompileSwift normal arm64 /Users/ffittschen/Dev/TemplateApp/Carthage/Checkouts/Eureka/Source/Cells.swift
    CompileSwift normal arm64 /Users/ffittschen/Dev/TemplateApp/Carthage/Checkouts/Eureka/Source/Rows.swift
    CompileSwift normal arm64 /Users/ffittschen/Dev/TemplateApp/Carthage/Checkouts/Eureka/Source/Controllers.swift
    CompileSwiftSources normal arm64 com.apple.xcode.tools.swift.compiler
(5 failures)
<unknown>:0: error: initializer requirement 'init' can only be satisfied by a `required` initializer in the definition of non-final class 'Section'
/Users/ffittschen/Dev/TemplateApp/Carthage/Checkouts/Eureka/Source/Core.swift:307:67: warning: '++' is deprecated: it will be removed in Swift 3
/Users/ffittschen/Dev/TemplateApp/Carthage/Checkouts/Eureka/Source/Core.swift:307:9: warning: C-style for statement is deprecated and will be removed in a future version of Swift
/Users/ffittschen/Dev/TemplateApp/Carthage/Checkouts/Eureka/Source/Core.swift:436:52: warning: '--' is deprecated: it will be removed in Swift 3
/Users/ffittschen/Dev/TemplateApp/Carthage/Checkouts/Eureka/Source/Core.swift:439:90: warning: '++' is deprecated: it will be removed in Swift 3
/Users/ffittschen/Dev/TemplateApp/Carthage/Checkouts/Eureka/Source/Core.swift:550:67: warning: '++' is deprecated: it will be removed in Swift 3
/Users/ffittschen/Dev/TemplateApp/Carthage/Checkouts/Eureka/Source/Core.swift:550:9: warning: C-style for statement is deprecated and will be removed in a future version of Swift
/Users/ffittschen/Dev/TemplateApp/Carthage/Checkouts/Eureka/Source/Core.swift:793:48: warning: '--' is deprecated: it will be removed in Swift 3
/Users/ffittschen/Dev/TemplateApp/Carthage/Checkouts/Eureka/Source/Core.swift:796:82: warning: '++' is deprecated: it will be removed in Swift 3
/Users/ffittschen/Dev/TemplateApp/Carthage/Checkouts/Eureka/Source/Helpers.swift:72:52: warning: 'init(start:end:)' is deprecated: it will be removed in Swift 3.  Use the '..<' operator.
<unknown>:0: error: initializer requirement 'init' can only be satisfied by a `required` initializer in the definition of non-final class 'Section'
A shell task failed with exit code 65:
2016-02-09 12:12:09.245 xcodebuild[24873:8586812]  DVTAssertions: Warning in /Library/Caches/com.apple.xbs/Sources/IDEFrameworks/IDEFrameworks-10160.3/IDEFoundation/Playgrounds/IDEPlaygroundAuxiliarySourceCompilerOperation.m:312
Details:  Unable to read diagnostics from file "/Users/ffittschen/Library/Developer/Xcode/DerivedData/Eureka-cxulzhapymzunnejqpbsnszfxwdk/Build/Intermediates/Eureka.build/Release-iphoneos/Eureka.build/Objects-normal/arm64/Rows.dia" (Invalid File): Invalid diagnostics signature
Function: void XCGenerateDiagnosticsFromFile(NSString *__strong, NSString *__strong, NSDictionary *__strong, NSDictionary *__strong, IDEActivityLogSectionRecorder *__strong, BOOL (^__strong)(IDEActivityLogMessage *__strong))
Thread:   <NSThread: 0x7fae66d222b0>{number = 6, name = (null)}
Please file a bug at http://bugreport.apple.com with this warning message and any useful information you can provide.
2016-02-09 12:12:09.846 xcodebuild[24873:8586812]  DVTAssertions: Warning in /Library/Caches/com.apple.xbs/Sources/IDEFrameworks/IDEFrameworks-10160.3/IDEFoundation/Playgrounds/IDEPlaygroundAuxiliarySourceCompilerOperation.m:312
Details:  Unable to read diagnostics from file "/Users/ffittschen/Library/Developer/Xcode/DerivedData/Eureka-cxulzhapymzunnejqpbsnszfxwdk/Build/Intermediates/Eureka.build/Release-iphoneos/Eureka.build/Objects-normal/arm64/Cells.dia" (Invalid File): Invalid diagnostics signature
Function: void XCGenerateDiagnosticsFromFile(NSString *__strong, NSString *__strong, NSDictionary *__strong, NSDictionary *__strong, IDEActivityLogSectionRecorder *__strong, BOOL (^__strong)(IDEActivityLogMessage *__strong))
Thread:   <NSThread: 0x7fae66d222b0>{number = 6, name = (null)}
Please file a bug at http://bugreport.apple.com with this warning message and any useful information you can provide.
** BUILD FAILED **


The following build commands failed:
    CompileSwift normal arm64 /Users/ffittschen/Dev/TemplateApp/Carthage/Checkouts/Eureka/Source/Core.swift
    CompileSwift normal arm64 /Users/ffittschen/Dev/TemplateApp/Carthage/Checkouts/Eureka/Source/Cells.swift
    CompileSwift normal arm64 /Users/ffittschen/Dev/TemplateApp/Carthage/Checkouts/Eureka/Source/Rows.swift
    CompileSwift normal arm64 /Users/ffittschen/Dev/TemplateApp/Carthage/Checkouts/Eureka/Source/Controllers.swift
    CompileSwiftSources normal arm64 com.apple.xcode.tools.swift.compiler
(5 failures)
```

**Comment by TeunVR** [comment]
Beta 4 still crashing :(

**Comment by jonnolen** [comment]
I was also getting prefer associatedtype over typealias warning and seg fault 11 when trying to build using xcode 7.3 beta (7D152p)

**Comment by mtnbarreto** [comment]
Guys, 
After some hard work Eureka finally compiles successfully on Xcode7.3 but not using the code7.3 beta 4 default toolchain. We have to use https://ci.swift.org/view/Packages/job/oss-swift-2.2-package-osx/ws/ toolchain and up. Probably it will work on next Xcode 7.3 Beta Default toolchain.

Master branch still has some swift2.2 warnings that will be there until Xcode7.3 is released. I think no one will use Xcode beta version for production apps. 

Anyway i'm gonna fix Xcode 7.3 warnings (like `prefer associatedtype over typealias`) here: https://github.com/xmartlabs/Eureka/tree/fix/212.

**Comment by Nick-The-Uncharted** [comment]
Add final to your custom cell solved it.

**Comment by jaz-ah** [comment]
unfortunately still crashing w/ the final released xcode7.3.  hopefully they will get the fix in a future version..

**Comment by cday** [comment]
I as well was hitting a segfault 11 with the final release of Xcode 7.3. Forunately, I was able to fix it by updating my pod file with `pod 'Eureka', :git => 'https://github.com/xmartlabs/Eureka.git', :branch => 'Xcode7_3'`. Seems like @mtnbarreto already fixed the issue and has yet to merge it and bump the pod version.

**Comment by mtnbarreto** [comment]
I think It was already fixed on master branch, anyway [Eureka 1.5.0](https://github.com/xmartlabs/Eureka/releases/tag/1.5.0) was just released. From now Eureka requires Xcode 7.3+.
